### PR TITLE
Feat: added mark notification as read endpoint

### PR DESCRIPTION
**Pull request by codewitgabi**
​

## Description

This endpoint adds a new endpoint `[PATCH] /api/v1/notifications/{id}` which marks a notification as read. The notification can only be marked as `read` if the user making the request is the owner of the notification with the given `id`.

​

## Related Issue (Link to issue ticket)

[Related issue](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/336)

## Screenshots (if appropriate - Postman, etc):
Screenshot 2024-07-24 142810
image

​

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/v1/test_notification.py
Comment (joboy-dev) [inline tests/v1/test_notification.py:1]: Can you update this test to mock the db
Check the user deactivation test for a sample

Good job so far

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes. Good job

**Comment by codewitgabi** [comment]
@joboy-dev

**Review by Goketech — APPROVED** [review]

**Comment by codewitgabi** [comment]
@Dev-wonderful

**Comment by codewitgabi** [comment]
@buka4rill please I need to know why my PR was closed🤔

**Comment by codewitgabi** [comment]
@buka4rill please I need to know why my PR was closed🤔

**Pull request closed without merging**